### PR TITLE
Cluster Overview page MesosStateStore refactoring

### PR DESCRIPTION
**Pull request by nLight**
This PR replaces MesosStateStore with direct calls to the V1 API. The API lacks `elapsed_time` and `start_time`, there's a ticket for that in mesos.

Closes DCOS-19145

### CODE REVIEW

File: src/js/pages/system/OverviewDetailTab.js [hunk 1]
```diff
@@ -220,14 +250,29 @@ class OverviewDetailTab extends mixin(StoreMixin) {
...
+    const mesosMasterInfo = masterInfo || this.getLoading();
+    const mesosCluster = cluster || this.getLoading();
+    const mesosVersion = version || this.getLoading();
+    const mesosBuilt = buildTime
+      ? moment(buildTime * 1000).fromNow()
+      : this.getLoading();
+    const mesosStarted = startTime
+      ? moment(startTime * 1000).fromNow()
+      : this.getLoading();
+    const mesosElected = electedTime
```
Comment (bstavroulakis) [inline src/js/pages/system/OverviewDetailTab.js]: When I ran this the started and elected time were endlessly loading. The info.start_time was null and I see that the electedTime is not set somewhere. Is this intended?

image

Comment (nLight) [reply src/js/pages/system/OverviewDetailTab.js]: Yep, as I pointed in the description, V1 API lacks elapsed_time and start_time, there's a ticket for that in mesos.

File: src/js/pages/system/OverviewDetailTab.js [hunk 2]
```diff
@@ -66,14 +71,39 @@ class OverviewDetailTab extends mixin(StoreMixin) {
...
         events: ["dcosBuildInfoChange", "dcosSuccess", "success"]
-      },
-      {
-        name: "state",
-        events: ["success"],
-        listenAlways: false
       }
     ];
 
+    request({ type: "GET_FLAGS" }).subscribe(message => {
```
Comment (MatApple) [inline src/js/pages/system/OverviewDetailTab.js]: @nLight wondering if we should have some error handling/retry here? Looks like we don't have any for the `mesos-client` `request` and `subscribe` methods.

**Comment by Poltergeist** [comment]
**Conflicting files**
npm-shrinkwrap.json
package.json

Please rebase.

File: src/js/pages/system/OverviewDetailTab.js (hunk 2 above)
Comment (nLight) [reply src/js/pages/system/OverviewDetailTab.js]: Do you think we should add like `retry(3)` just to be on a safe side?

**Comment by nLight** [comment]
@bstavroulakis @MatApple @Poltergeist I rebased the branch, please re-review 🙇

File: src/js/pages/system/OverviewDetailTab.js (hunk 2 above)
Comment (MatApple) [reply src/js/pages/system/OverviewDetailTab.js]: I think we should and we should really show an error message after that. We need to architect our request error handling a little e.g. how we deal with 401, 403 vs 500's. A retry on 403 is not smart. cc/ @orlandohohmeier

Comment (MatApple) [reply src/js/pages/system/OverviewDetailTab.js]: BTW - I don't think it should block this PR.

Comment (nLight) [reply src/js/pages/system/OverviewDetailTab.js]: 👍  https://jira.mesosphere.com/browse/DCOS-19443

**Pull request merged**